interface/java_binding/src/main/java/org/khronos/ktx/KTXAstcParams.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021, Shukant Pal and Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.khronos.ktx;
+
+public class KTXAstcParams {
+    private boolean verbose;
+    private int threadCount;
+    private int blockDimension;
+    private int mode;
+    private int qualityLevel;
+    private boolean normalMap;
+    private char[] inputSwizzle = new char[4];
+
```

Comment:
I don't have an opinion on this. It seems like a conventional Java pattern. See this example of [KafkaApiBuilder](https://github.com/apache/kafka/blob/db42afd6e24ef4291390b4d1c1f10758beedefed/core/src/main/java/kafka/server/builders/KafkaApisBuilder.java#L44) in the Apache Kafka project. I'd be down to stripping out the getters, however, since I don't think people would need to get back the retrieved values.